great_expectations/expectations/expectation.py

```diff
@@ -1592,6 +1592,51 @@ def _validate_metric_value_between(  # noqa: PLR0912
         return {"success": success, "result": {"observed_value": metric_value}}
 
 
+class SqlExpectation(BatchExpectation, ABC):
+    # TODO: Need to rename to unexpected_rows_query (and make corresponding metric change)
+    query: str
+
+    metric_dependencies: ClassVar[Tuple[str, ...]] = ("query.table",)
+    success_keys: ClassVar[Tuple[str, ...]] = ("query",)
+    domain_keys: ClassVar[Tuple[str, ...]] = (
+        "batch_id",
+        "row_condition",
+        "condition_parser",
+    )
+
+    @pydantic.validator("query")
+    def _validate_query(cls, query: str) -> str:
+        is_valid = "active_batch" in query
+        try:
+            # Using a dummy value to check syntactic correctness
+            substituted_query = query.format(active_batch="dummy_value")
+            # If the query is not parameterized, it will be the same as the substituted query
+            if query == substituted_query:
+                is_valid = False
+        except KeyError:
+            is_valid = False
```

Comment:
In the import section do `from string import Formatter` then here do:
```
is_valid = "active_batch" in [f[1] for f in Formatter().parse(query)]
```
See docs [here](https://docs.python.org/3/library/string.html#string.Formatter.parse). Basically python needs to be able to parse format strings and provides the same capability to developers. You can iterate and print the elements of `Formatter().parse(query)` to see what it returns. Basically the first element is a string with the text before a format variable `{name}` and the second element (that I grab in the snippet above) is the `name` of the format variable.